cli/file_watcher.rs

```diff
@@ -11,14 +12,81 @@ use notify::Error as NotifyError;
 use notify::RecommendedWatcher;
 use notify::RecursiveMode;
 use notify::Watcher;
+use std::mem;
 use std::path::PathBuf;
 use std::pin::Pin;
+use std::sync::{mpsc, mpsc::Receiver};
+use std::time::{Duration, Instant};
 use tokio::select;
-use tokio::sync::mpsc;
+
+const DEBOUNCE_TIMEOUT_MS: Duration = Duration::from_millis(200);
 
 // TODO(bartlomieju): rename
 type WatchFuture = Pin<Box<dyn Future<Output = Result<(), AnyError>>>>;
 
+struct Debounce {
+  rx: Receiver<Result<NotifyEvent, AnyError>>,
+  debounce_time: Duration,
+  start_time: Option<Instant>,
+  last_event: Option<NotifyEvent>,
+}
+
+impl Debounce {
+  fn new(
+    rx: Receiver<Result<NotifyEvent, AnyError>>,
+    debounce_time: Duration,
+  ) -> Self {
+    Self {
+      rx,
+      debounce_time,
+      start_time: None,
+      last_event: None,
+    }
+  }
+}
+
+impl Stream for Debounce {
+  type Item = NotifyEvent;
+
+  fn poll_next(
+    self: Pin<&mut Self>,
+    cx: &mut Context,
+  ) -> Poll<Option<Self::Item>> {
+    let inner = self.get_mut();
+    if let Ok(Ok(event)) = inner.rx.try_recv() {
+      match inner.last_event.as_ref() {
+        Some(last_event) if *last_event == event => {
+          // if received event is the same as previous one, reset timeout
+          inner.start_time = Some(Instant::now());
+        }
```

Comment:
I don't understand this bit, is it possible to receive exactly same event twice?

Reply:
It's based on @bartOssh's code: https://github.com/denoland/deno/blob/988e6c19386a15543a09044a483cbca6aeeff38b/cli/file_watcher.rs#L58-L61

But after thinking again about it, this check seems unnecessary - if any event is received from `rx`, we always should reset timeout. 
I'll fix the code.